src/users/service/UsersService.ts

```diff
@@ -0,0 +1,36 @@
+import RequestClientWrapper from '../../RequestClientWrapper';
+import GenerateApiError from '../../utils/GenerateApiError';
+import * as Q from 'q';
+import Version from '../../model/Version';
+import Resource from '../../model/Resource';
+import User from '../model/User';
+import { AxiosResponse } from 'axios';
+
+export default class UsersService {
+
+  private readonly EXPECTED_GET_STATUS = 200;
+
+  constructor(private readonly client: RequestClientWrapper) { }
+
+  public get(userId: string): Promise<User> {
+    const deferred = Q.defer<User>();
+    const requestUrl = `/${Version.V1}${Resource.USERS}/${userId}`;
+    this.client.get(requestUrl, {}, (error, response, body) => {
+      this.mapUserPromise(requestUrl, deferred, error, response, body, this.EXPECTED_GET_STATUS);
+    });
+    return deferred.promise as any;
+  }
+
+  private mapUserPromise(requestUrl: string, deferred: Q.Deferred<User>, error: any, response: AxiosResponse,
```

Comment:
This is in like every service. It's not really specific to the type being returned, so we can probably extract this iinto it's own service leveraging generic types, since the only thing that is different is the deferred argument type. Might have to create a parameter object as well since we are getting up there in number of arguments.

    private mapUserPromise(requestUrl: string, deferred: Q.Deferred<T>, error: any, response: AxiosResponse,
      body: any, expectedStatusCode: number) {
      if (error) {
        deferred.reject(error);
      } else if (response.status === expectedStatusCode) {
        deferred.resolve(body);
      } else {
        const channelApeErrorResponse = GenerateApiError(requestUrl, response, body, expectedStatusCode);
        deferred.reject(channelApeErrorResponse);
      }
    }

At the very least here you should either remove the argument of expected status code and use the constant everywhere in this method or only use the argument(even when creating API error)

Reply:
@ryankazokas I pulled the mapping method into its own service which the service classes can extend.  Let me know what you think